src/Consumer/ReferentMessageDispatcherConsumer.php

```diff
@@ -85,6 +82,29 @@ public function doExecute(array $data): int
         }
     }
 
+    public function sendMessage(ReferentManagedUsersMessage $savedMessage, ReferentMessage $message, array $recipients, int $count)
+    {
+        $this->getDoctrine()->getConnection()->beginTransaction();
+
+        try {
+            $this->mailer->sendMessage(MailjetMessage::createFromModel($message, $recipients));
+
+            $this->writeln(
+                $savedMessage->getUuid()->toString(),
+                'Message from '.$message->getFrom()->getEmailAddress().' dispatched ('.$count.')'
+            );
+
+            $this->referentMessageRepository->incrementOffset($savedMessage, count($recipients));
```

Comment:
I would remove this of the try, otherwise this line still can be by-passed:
```php
try {
    $this->mailer->sendMessage(MailjetMessage::createFromModel($message, $recipients));

    $this->writeln(
        $savedMessage->getUuid()->toString(),
        'Message from '.$message->getFrom()->getEmailAddress().' dispatched ('.$count.')'
    );

    $this->getDoctrine()->getConnection()->commit();
} catch (\Throwable $error) {
    $this->getManager()->close();
    $this->getDoctrine()->getConnection()->rollback();
} finally {
    $this->referentMessageRepository->incrementOffset($savedMessage, count($recipients));
    $this->getDoctrine()->getManager()->flush();

    if (isset($error)) {
        throw $error;
    }
}
```
Consider keeping the reference of the connection to remove some calls.